llvm/include/llvm/TextAPI/SymbolSet.h

```diff
@@ -28,6 +29,18 @@ struct SymbolsMapKey {
 
   SymbolsMapKey(MachO::EncodeKind Kind, StringRef Name)
       : Kind(Kind), Name(Name) {}
+
+  bool operator==(const SymbolsMapKey &Other) const {
+    return Kind == Other.Kind && Name == Other.Name;
+  }
+
+  bool operator!=(const SymbolsMapKey &Other) const {
+    return operator==(Other);
+  }
+
+  bool operator<(const SymbolsMapKey &Other) const {
+    return Kind < Other.Kind || Name < Other.Name;
```

Comment:
```suggestion
    return std::make_pair(Kind, Name) < std::make_pair(Other.Kind, Other.Name);
```

If `Kind > Other.Kind`, this will confusingly return `true` if `Name < Other.Name`.